snap/implicit.go

```diff
@@ -79,3 +85,54 @@ func makeImplicitSlot(snapInfo *Info, ifaceName string) *SlotInfo {
 		Interface: ifaceName,
 	}
 }
+
+// addImplicitHooks adds hooks from the installed snap's hookdir to the snap info.
+//
+// Existing hooks (i.e. ones defined in the YAML) are not changed; only missing
+// hooks are added.
+func addImplicitHooks(snapInfo *Info) error {
+	// First of all, check to ensure the hooks directory exists. If it doesn't,
+	// it's not an error-- there's just nothing to do.
+	hooksDir := snapInfo.HooksDir()
+	if _, err := os.Stat(hooksDir); os.IsNotExist(err) {
+		return nil
+	}
+
+	fileInfos, err := ioutil.ReadDir(hooksDir)
+	if err != nil {
+		return fmt.Errorf("unable to read hooks directory: %s", err)
+	}
+
+	for _, fileInfo := range fileInfos {
+		addHookName(snapInfo, fileInfo.Name())
+	}
+
+	return nil
+}
+
+// addImplicitHooksFromContainer adds hooks from the snap file's hookdir to the snap info.
+//
+// Existing hooks (i.e. ones defined in the YAML) are not changed; only missing
+// hooks are added.
+func addImplicitHooksFromContainer(snapInfo *Info, snapf Container) error {
```

Comment:
This function exists in order to reuse the [existing validation that happens before the container is mounted](https://github.com/snapcore/snapd/pull/1329/files#diff-adf1cc1411235331ed9a1d6516eef14aL372). We could do it later, but then we'd have to validate twice.